model/jpa/src/main/java/org/keycloak/events/jpa/JpaEventStoreProviderFactory.java

```diff
@@ -30,15 +30,18 @@
 public class JpaEventStoreProviderFactory implements EventStoreProviderFactory {
 
     public static final String ID = "jpa";
+    private int maxDetailLength;
 
     @Override
     public EventStoreProvider create(KeycloakSession session) {
         JpaConnectionProvider connection = session.getProvider(JpaConnectionProvider.class);
-        return new JpaEventStoreProvider(connection.getEntityManager());
+        return new JpaEventStoreProvider(connection.getEntityManager(), maxDetailLength);
     }
 
     @Override
     public void init(Config.Scope config) {
+        Integer maxDetail = config.getInt("max-detail-length", 0);
+        maxDetailLength = (maxDetail == null) ? 0 : maxDetail;
```

Comment:
maxDetail won't be null, so you can just do:
`maxDetailLength = config.getInt("max-detail-length", 0);`